### PR TITLE
Add outdoor POI support and UI components

### LINKED ISSUE #337: [Task 5.1.2] Filter & Show Outdoor POIs
### Description

Filter & Show Outdoor POIs from Google Places (already implemented)

Be able to distinguish different types of POIs

### PR DESCRIPTION
**Pull request by kcrri**
## What does this PR do?
Introduce outdoor points-of-interest support: add outdoorPOI data and category list, distance utilities (calculate/format/find nearest/in-range), and POI marker utilities (icons, labels, colors, SVG markers). Add UI components for POIs: POIInfoPanel, POIFilter, NearestPOIBanner, and tests for distance/poi marker/filter/info/banner. Integrate POI state and behavior into MapScreen (filtering, nearest calculation, panel animation, set-as-destination flow) and update BuildingInfo to allow setting a building as destination when user location is available. This change adds end-to-end POI functionality (data, utils, UI, and tests) to the mobile map feature.

## How to test
- [ ] Unit tests: npm test
- [ ] Test Coverage: npm test (or npm run test:coverage)

## Linked issue
Closes #337 #338 #339

## Review checklist
- [ ] code is readable
- [ ] tests updated/added
- [ ] edge cases handled

### CODE REVIEW

File: mobile/src/screens/MapScreen.tsx [hunk 1]
```diff
@@ -236,6 +252,35 @@ export default function MapScreen() {
...
+        duration: 250,
+        useNativeDriver: true,
+      }).start();
+    }
+  }, [selectedPOI]);
+
+  // Calculate nearest POI when user location changes
+  useEffect(() => {
+    const filteredPOIsForCampus = getFilteredPOIs();
+    const nearest = findNearestPOI(userLocation, filteredPOIsForCampus, 500);
```
Comment (Dopyo) [inline mobile/src/screens/MapScreen.tsx]: Instead of hardcoding 500 meter range, you could make this a configurable setting in SettingsScreen.

File: mobile/src/utils/distanceUtils.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+};
+
+/**
+ * Finds the nearest POI within a specified range
+ * @param userCoordinates User's current location
+ * @param pois Array of POIs to search
+ * @param rangeInMeters Maximum distance in meters (default: 500)
+ * @returns POI with distance info if found, null otherwise
+ */
+export const findNearestPOI = (
```
[the remark this answers is not in the document]
Comment (kcrri) [reply mobile/src/utils/distanceUtils.ts]: Done! I've refactored findNearestPOI to use findPOIsInRange and return the first element. This eliminates the duplicate logic for calculating distances and filtering by range. 

```suggestion
export const findNearestPOI = (
  userCoordinates: { latitude: number; longitude: number } | null,
  pois: Array<{ coordinates: { latitude: number; longitude: number }; [key: string]: any }>,
  rangeInMeters = 500
): (typeof pois[0] & { distance: number }) | null => {
  const poisInRange = findPOIsInRange(userCoordinates, pois, rangeInMeters);
  return poisInRange.length > 0 ? poisInRange[0] : null;
};

```

**Review by Dopyo — APPROVED** [review]
Approved
